metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -1367,6 +1367,28 @@ abstract class MetalsLspService(
       .asJavaObject
   }
 
+  def copyFQNOfSymbol(
+      params: l.TextDocumentPositionParams
+  ): CompletableFuture[Object] = {
+    Future.successful {
+      val path = params.getTextDocument.getUri.toAbsolutePath
+      val dialect = scalaVersionSelector.getDialect(path)
+      val pos = params.getPosition
+      Mtags
+        .index(path, dialect)
+        .occurrences
+        .filter(_.range.exists(_.encloses(pos)))
+        .map(_.symbol.symbolToFullQualifiedName)
+        .headOption
+    }.asJavaObject
+    // TODO(kπ) for methods with overloads, we need types of arguments, so need to call pc
```

Comment:
Do we need that? Also, we could use .info method maybe?

Reply:
For overloaded methods, we need to provide argument types for scaladoc, as per: https://www.baeldung.com/java-method-in-javadoc
@kasiaMarek suggested, that we can detect in mtags if there are any overloads and only use pc if we have to.

Which `.info` method do you mean? Is there `info` on an mtags symbol?

Reply:
There is info in PresentationCompiler interface, which might be useful

Reply:
Actually, before using Mtags you can use SemanticDB (should be faster in most cases). DefinitionProvider has method `symbolOccurrence` which finds occurrence in semanticdb and even has a fallback to Mtags (though you might want to add another one when no semanticdb is found).

For overloaded methods, you can find argument types in SemanticDB too (`SymbolInformation#signature`). But a fallback to pc might be unavoidable if you want it to work for e.g. methods created after the code stopped compiling.